### PR TITLE
expose RACDelegateProxy.h

### PR DESCRIPTION
**Pull request by SKIT1**

### CODE REVIEW

File: ReactiveCocoa.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2944,11 +2946,12 @@
 			baseConfigurationReference = D047263419E49FE8006002AA /* iOS-Framework.xcconfig */;
 			buildSettings = {
 				CODE_SIGN_IDENTITY = "iPhone Developer";
-				"CODE_SIGN_IDENTITY[sdk=iphonesimulator*]" = "";
+				"CODE_SIGN_IDENTITY[sdk=iphonesimulator*]" = "iPhone Developer";
```
Comment (mdiep) [inline ReactiveCocoa.xcodeproj/project.pbxproj]: Some unnecessary changes (like this one) were made to the project file. Please undo these changes.

**Comment by mdiep** [comment]
Can you provide some detail about why you'd like this class to be public?

This branch also has some conflicts that need to be resolved.

**Comment by SKIT1** [comment]
I think there is a need to expose RACDelegateProxy. h header file.For example, extending the delegate proxy.

struct AssociationKey {
 static var delegateProxy: UInt8 = 0
 }
extension UISearchBar {
	private func RACUseDelegateProxy(_self: UISearchBar) {
		if let delegate = _self.delegate as? RACDelegateProxy {
			if delegate == _self.rac_delegateProxy {
				return
			}
		}
		_self.rac_delegateProxy.rac_proxiedDelegate = _self.delegate
		_self.delegate = _self.rac_delegateProxy as? UISearchBarDelegate
	}

	public var rac_delegateProxy: RACDelegateProxy {

		return lazyAssociatedProperty(self, key: &AssociationKey.delegateProxy, factory: { () -> RACDelegateProxy in
			return RACDelegateProxy(withProtocol: UISearchBarDelegate.self)
		})
	}

	public var rac_textSignal: SignalProducer<String?, NSError> {
		let triggerSignal = toVoidSignal(self.rac_willDeallocSignal().toSignalProducer())

		let signal = SignalProducer<AnyObject?, NSError> { [unowned self] observer, _ in
			observer.sendNext(self)
			observer.sendCompleted()
		}.concat((self.rac_delegateProxy.rac_signalForSelector(#selector(UISearchBarDelegate.searchBar(_: textDidChange:))).toSignalProducer())).map({ (next) -> String? in
			if let tuple = next as? RACTuple {
				guard let searchBar = tuple.first as? UISearchBar else {
					return nil
				}
				return searchBar.text
			} else {
				guard let searchBar = next as? UISearchBar else {
					return nil
				}
				return searchBar.text
			}
		}).takeUntil(triggerSignal)

		RACUseDelegateProxy(self)

		return signal
	}
}

File: Carthage/Checkouts/Quick [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 2f037560be197f0f5ae992512549bc29fabb3818
+Subproject commit dc2b4b75d2f718518acbdef289a72fad77975a8a
```
Comment (mdiep) [inline Carthage/Checkouts/Quick]: Can you remove the change to the Quick version in this PR? I think that will resolve the conflict and make it so I can merge the PR.

Comment (mdiep) [inline Carthage/Checkouts/Quick]: The submodule commit needs to be reverted as well.

(Please post a comment when you're done so I get a notification!)

**Comment by mdiep** [comment]
Closing in favor of #2869.
